### PR TITLE
chore(kuma-cp) add customizable API

### PR DESCRIPTION
**Pull request by nickolaev**
### Summary

Adding customizable API.

### Documentation

- [ ] Not needed

### CODE REVIEW

File: pkg/api-server/customization/api_manager.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func (c *APIList) Add(ws *restful.WebService) {
+	c.list = append(c.list, ws)
+}
+
+func (c *APIList) AddRouteFunction(path, subpath string, handler restful.RouteFunction) {
+	ws := new(restful.WebService).Path(path)
+	ws.Route(ws.GET(subpath).To(handler))
+	c.Add(ws)
+}
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/customization/api_manager.go:35]: What will happen if you need something like Auth on this endpoint or filter of logging requests or add an extra method to this webservice?

I think I'd drop this method and do these 3 lines on the APIManager client that will just use `Add`. That's not really extensible API, you cannot add filters, change the method, add many methods to ws, add OpenAPI docs etc.

File: pkg/api-server/customization/api_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		Expect(err).ToNot(HaveOccurred())
+		wsManager := customization.NewAPIList()
+
+		wsManager.AddRouteFunction("foo", "baz", func(request *restful.Request, response *restful.Response) {
+			_ = response.WriteAsJson("bar")
+		})
+
+		apiServer := createTestApiServer(resourceStore, cfg, true, metrics, wsManager)
+
+		stop := make(chan struct{})
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/customization/api_manager_test.go:39]: close the channel

File: pkg/api-server/customization/customization_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		wsManager = customization.NewAPIList()
+	}
+	cfg := kuma_cp.DefaultConfig()
+	cfg.ApiServer = config
+	apiServer, err := api_server.NewApiServer(resources, wsManager, defs, &cfg, enableGUI, metrics)
+	Expect(err).ToNot(HaveOccurred())
+	return apiServer
+}
+
+var SampleTrafficRouteWsDefinition = definitions.ResourceWsDefinition{
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/customization/customization_suite_test.go:57]: I don't think we need this for a test

File: pkg/api-server/server.go [hunk 4]
```diff
@@ -71,7 +73,8 @@ func init() {
 	}
 }
 
-func NewApiServer(resManager manager.ResourceManager, defs []definitions.ResourceWsDefinition, cfg *kuma_cp.Config, enableGUI bool, metrics metrics.Metrics) (*ApiServer, error) {
+func NewApiServer(resManager manager.ResourceManager, wsManager customization.APIInstaller,
+	defs []definitions.ResourceWsDefinition, cfg *kuma_cp.Config, enableGUI bool, metrics metrics.Metrics) (*ApiServer, error) {
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/server.go:77]: formatting nit. For me, it's easier to read arguments grouped in one line or every in it's own line so either
```
func NewApiServer(resManager manager.ResourceManager, wsManager customization.APIInstaller, defs []definitions.ResourceWsDefinition, cfg *kuma_cp.Config, enableGUI bool, metrics metrics.Metrics) (*ApiServer, error) {
```
or
```
func NewApiServer(
    resManager manager.ResourceManager,
    wsManager customization.APIInstaller,
    defs []definitions.ResourceWsDefinition,
    cfg *kuma_cp.Config,
    enableGUI bool,
    metrics metrics.Metrics,
) (*ApiServer, error) {
```

File: pkg/api-server/server.go [hunk 5]
```diff
@@ -135,6 +138,10 @@ func NewApiServer(resManager manager.ResourceManager, defs []definitions.Resourc
 	} else {
 		container.ServeMux.HandleFunc("/gui/", newApiServer.notAvailableHandler)
 	}
+
+	// Install the custom WS
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/server.go:142]: nit: I don't think this comment is needed. I like commenting out code for the following reasons
1) explaining _why_ something is happening
2) explaining code design and design decisions / tradeoffs
3) explaining difficult code when language is crippling the developer to write an easy to understand code

Here it's pretty obvious what is going on

**Review by jakubdyszkiewicz — COMMENTED** [review]
Nice!

**Comment by jakubdyszkiewicz** [comment]
it's `chore(kuma-cp)`

File: pkg/api-server/customization/api_manager.go (hunk 1 above)
Comment (nickolaev) [reply pkg/api-server/customization/api_manager.go:35]: This is just a convenience method, some will prefer this for simplicity and some to supply a full-featured WS. I don't see what is wrong with having this one here.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**